site/src/pages/release-notes/1.21.0.mdx

```diff
@@ -0,0 +1,148 @@
+
+  Server
+    .builder()
+    .setHeader(CUSTOM_HEADER, "server-default-header")
+    .virtualHost("foo.com")
+    .setHeader(CUSTOM_HEADER, "vhost-default-header").and()
+    .route()
+    .path("/foo")
+    .setHeader(CUSTOM_HEADER, "service-default-header")
+  ```
+- You can now set <type://Endpoint#host()> and `:authority` or `Host` header
+  differently using <type://ClientBuilder#setHeader(CharSequence,Object)>. #4440 #4520
+  ```java 
+  WebClient
+    // A request is sent to 'reverse-proxy.example.com'.
+    .builder("http://reverse-proxy.example.com/")
+    // but 'my-order.service.com' is used for the ':authority' header.
+    .setHeader(HeaderNames.AUTHORITY, "my-order.service.com")
+    ...
+  ```
+- <type://CircuitBreakerClient> can now be used with arbitrary <type://CircuitBreaker> implementations. #4426
+- Added <type://Bytes> that represents binary data. #4505
+- You can now see the description of services in the overview of <type://DocService>. #4480 #4537
+- You can now override an authority of a gRPC client using `CallOptions`. #4445 #4520
+
+## 📈 Improvements
+
+- <type://DecodingClient> and <type://DecodingService> can now limit the maximum length of decompressed
+  data. #4469 #4564
+- Replace `synchronized` blocks with <type://ReentrantLock> to make virtual thread friendly. #4551 #4549
+- <type://ServiceSpecification> for <type://DocService> are lazily loaded so as not to block a <type://Server>
+  from starting. #4491
+
+## 🛠️ Bug fixes
+
+- <type://HttpHeaders> set by <type://ClientBuilder#setHeader(CharSequence,Object)>,
+  <type://ClientBuilder#setHeader(CharSequence,Object)> and their variants are now stored in
+  <type://ClientRequestContext#defaultRequestHeaders()> which have the lowest priority and don't override the
+  values in <type://RequestHeaders>.
+  - Previously, the <type://HttpHeaders> set via <type://ClientBuilder> were stored in
+    <type://ClientRequestContext#additionalRequestHeaders()> which may override the values in
+    <type://RequestHeaders> #4440 #4520
+- Fixed a regression that `void` methods returns `200 OK` instead of `204 NoContent` if <type://@Produces> 
+  annotation is added. #4575
+- Deleted uncompressed <type://DocService> assets that were added unintentionally. #4500 #4568
+- An exception raised in <type://Authorizer> is now properly peeled using
+  <type://Exceptions#peel(Throwable)>. #4567
+- You can now use the exception, which is raised in the decorator, to make a decision for retrying. #4559
+- <type://EndpointGroup> is now correctly copied to a newly derived client. #4556
+- <type://RequestContextExportingAppender> can now be used with logback `SocketAppender`. #4546 #4547
+- Reversed HTTP/1.1 responses are now correctly ordered even when HTTP/1.1 pipelining is enabled. #4536 #4470
+- `414 Request-URI Too Long` is now correctly returned for a long URI in HTTP/1.1. #4478 #4532
+- Fixed to propagate the cause of a decoder failure of an HTTP/1 request to
```

Comment:
```suggestion
- Fixed to propagate the cause of a decoder failure of an HTTP/1.1 request to
```
or
`an HTTP/1.0 or HTTP/1.1 request`

Reply:
Let me just use HTTP/1.1 since we don't officially support HTTP/1.0.